metadata-ws/src/test/resources/features/sample.feature

```diff
@@ -0,0 +1,340 @@
+    And the response should contain field name with value test_human_sample
+    When I request GET for taxonomies of SAMPLE
+    Then the response code should be 200
+    And the response should contain one taxonomy
+    And the href of the taxonomy of taxonomies has items TAXONOMY
+
+    When I request PATCH SAMPLE with list <list> of taxonomies
+    Then the response code should be 4xx
+    And the response should contain field exception with value <exception>
+
+    Examples:
+      | list | exception |
+      | NONE | uk.ac.ebi.ampt2d.metadata.exceptionhandling.SampleWithoutTaxonomyException |
+      | BLANK | uk.ac.ebi.ampt2d.metadata.exceptionhandling.InvalidTaxonomyException |
+      | TAXONOMY,BLANK | uk.ac.ebi.ampt2d.metadata.exceptionhandling.InvalidTaxonomyException |
+
+
+  Scenario Outline: find two samples by taxonomy
+    When I request POST /taxonomies with JSON payload:
+    """
+    {
+      "taxonomyId": 207598,
+      "name": "Homininae"
+    }
+    """
+    Then set the URL to TAXONOMY_1
+    When I request POST /taxonomies with JSON payload:
+    """
+    {
+      "taxonomyId": 9606,
+      "name": "Homo Sapiens"
+    }
+    """
+    Then set the URL to TAXONOMY_2
+    When I create a parameterized sample with Species1 for accession, 1 for version, Species collection1 for name and TAXONOMY_1,TAXONOMY_2 for taxonomy
+    Then the response code should be 201
+    And set the URL to SAMPLE1
+    When I request POST /taxonomies with JSON payload:
+    """
+    {
+      "taxonomyId": 9597,
+      "name": "Pan paniscus"
+    }
+    """
+    Then set the URL to TAXONOMY_3
+    When I request POST /taxonomies with JSON payload:
+    """
+    {
+      "taxonomyId": 9598,
+      "name": "Pan troglodytes"
+    }
+    """
+    Then set the URL to TAXONOMY_4
+    When I create a parameterized sample with Species2 for accession, 1 for version, Species collection2 for name and TAXONOMY_3,TAXONOMY_4 for taxonomy
+    Then the response code should be 201
+    And set the URL to SAMPLE2
+    When I create a parameterized sample with Species3 for accession, 1 for version, Species collection3 for name and TAXONOMY_1,TAXONOMY_3 for taxonomy
+    Then the response code should be 201
+    And set the URL to SAMPLE3
+
+    When I request search for the samples with the parameters: <query>
+    Then the response code should be 200
+    And the response should contain 2 samples
+    And the href of the sample of samples has items <url>
+
+    Examples:
+      | query | url |
+      | taxonomies.name=Homininae | SAMPLE1,SAMPLE3 |
+      | taxonomies.name=Pan paniscus | SAMPLE2,SAMPLE3 |
+      | taxonomies.taxonomyId=207598 | SAMPLE1,SAMPLE3 |
+      | taxonomies.taxonomyId=9597 | SAMPLE2,SAMPLE3 |
+
+
+  Scenario Outline: find one sample by taxonomy
+    When I request POST /taxonomies with JSON payload:
+    """
+    {
+      "taxonomyId": 207598,
+      "name": "Homininae"
+    }
+    """
+    Then set the URL to TAXONOMY_1
```

Comment:
Why not using valid URLs in all these examples? The system should in fact complain if something like "TAXONOMY_1" is provided.

Reply:
It was designed to store name with actual URL in map. The same was continued.
Otherwise we need to extract the URL and pass in next statements